client/src/portal/api/mod.rs

```diff
@@ -252,6 +252,17 @@ impl Keyring {
             self.iteration_count.try_into().unwrap(),
         )
     }
+
+    pub(crate) fn reset_salt(&mut self) {
+        self.salt = rand::thread_rng().gen::<[u8; DEFAULT_SALT_SIZE]>().to_vec();
+    }
+
+    pub(crate) fn set_modified_time(&mut self) {
```

Comment:
Please merge both methods.

Reply:
And also reset iteration_count/usage_count, the salt_len also needs to be updated.

Reply:
Introduced `reset()` to update/reset all the Keyring fields: https://github.com/bilelmoussaoui/oo7/pull/77/commits/4a97e2f129d304a96cbe2a0dfb3372f40a0b6a05